events/nats_message.go

```diff
@@ -0,0 +1,217 @@
+package events
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/nats-io/nats.go"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/propagation"
+)
+
+func natsSubscriptionMessageChan[T any](conn *NatsConnection, batchSize int, natsCh <-chan *nats.Msg) chan Message[T] {
+	msgCh := make(chan Message[T], batchSize)
+
+	go func() {
+		for nMsg := range natsCh {
+			msg := natsDecodeMessage[T](conn, nMsg)
+
+			msgCh <- msg
+		}
+
+		close(msgCh)
+	}()
+
+	return msgCh
+}
+
+func natsSubscriptionAuthRelationshipRequestChan(conn *NatsConnection, batchSize int, natsCh <-chan *nats.Msg) chan Request[AuthRelationshipRequest, AuthRelationshipResponse] {
+	msgCh := make(chan Request[AuthRelationshipRequest, AuthRelationshipResponse], batchSize)
+
+	go func() {
+		for nMsg := range natsCh {
+			msg := natsDecodeMessage[AuthRelationshipRequest](conn, nMsg)
+
+			msgCh <- &NatsAuthRelationshipRequest{NatsMessage: msg.(*NatsMessage[AuthRelationshipRequest])}
```

Comment:
Same thoughts as before re: `select` for these send operations. It's possible the context for a method like `SubscribeAuthRelationshipRequests` could be canceled, causing this send operation to hang forever.

Reply:
ah yes, easy fix.